FileShareService.DesktopClient.Core/JobsParser.cs

```diff
@@ -26,12 +35,144 @@ public Jobs.Jobs Parse(string jobs)
+
+                    switch (jobAction)
+                    {
+                        case NewBatchJob.JobAction:
+                            job = JsonConvert.DeserializeObject<NewBatchJob>(jsonString, jsonSerializerSettings);
+                            break;
+
+                        case AppendAclJob.JobAction:
+                            job = JsonConvert.DeserializeObject<AppendAclJob>(jsonString, jsonSerializerSettings);
+                            break;
+
+                        case SetExpiryDateJob.JobAction:
+                            job = JsonConvert.DeserializeObject<SetExpiryDateJob>(jsonString, jsonSerializerSettings);
+                            break;
+                    }
+
+                    if (job != null)
+                    {
+                        job.Validate(batchJob);
+                        jobCollection.Add(job);
+                    }
+                }
+
+                //Add error job at begining of the collection, if exists.
+                if (errorJob != null)
+                {
+                    jobCollection.Insert(0, errorJob);
+                }
+
+                return  new Jobs.Jobs() { jobs = jobCollection};
             }
             catch (Exception e)
             {
-                return new Jobs.Jobs() {jobs = new[] {new ErrorDeserializingJobsJob(e)}};
+                return new Jobs.Jobs() { jobs = new[] { new ErrorDeserializingJobsJob(e) { ErrorMessages = new List<string>() { e.Message } } } };
+            }
+        }
+
+        private List<string> ValidateJobActionAndDisplayName(JToken job, out string jobAction)
+        {
+            jobAction = string.Empty;
+
+            List<string> errors = new List<string>();
+
+            //Retrieve job action
+            JToken? jobActionToken = job.SelectToken("action");
+
+            if (jobActionToken == null ||
+                string.IsNullOrWhiteSpace(Convert.ToString(jobActionToken)))
+            {
+                errors.Add(AddLineInfo(job, "Job action is not specified or is invalid."));
+                return errors;
             }
+
+            jobAction = $"{Convert.ToString(jobActionToken)}";
+
+            //Check whether job action specified in config is valid or not
+            if (!ValidJobActions.Contains(jobAction))
+            {
+                errors.Add(AddLineInfo(job, $"Specified job action '{jobAction}' is invalid."));
+                return errors;
+            }
+
+            string displayName = string.Empty;
+            //Retrieve display name
+            JToken? displayNameTokne = job.SelectToken("displayName");
+            if (displayNameTokne == null || string.IsNullOrWhiteSpace(Convert.ToString(displayNameTokne)))
+            {
+                errors.Add(AddLineInfo(job, "Job display name is not specified or is invalid."));
+                return errors;
+            }
+
+            displayName = displayNameTokne.ToString();
```

Comment:
Please combine the declaration and initialisation of `displayName`. Please also use `Convert.ToString()` for consistency.
````
var displayName = Convert.ToString(displayNameTokne);
````

Reply:
Done